crates/rethnet_eth/src/transaction/request/eip4844.rs

```diff
@@ -0,0 +1,104 @@
+use std::sync::OnceLock;
+
+use bytes::Bytes;
+use revm_primitives::{keccak256, Address, B256, U256};
+use secp256k1::SecretKey;
+
+use crate::{
+    access_list::AccessListItem, signature::Signature, transaction::Eip4844SignedTransaction,
+    utils::envelop_bytes,
+};
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[cfg_attr(
+    feature = "fastrlp",
+    derive(open_fastrlp::RlpEncodable, open_fastrlp::RlpDecodable)
+)]
+pub struct Eip4844TransactionRequest {
+    pub chain_id: u64,
+    pub nonce: u64,
+    pub max_priority_fee_per_gas: U256,
+    pub max_fee_per_gas: U256,
+    pub max_fee_per_blob_gas: U256,
+    pub gas_limit: u64,
+    pub to: Address,
+    pub value: U256,
+    pub input: Bytes,
+    pub access_list: Vec<AccessListItem>,
+    pub blob_hashes: Vec<B256>,
+}
+
+impl Eip4844TransactionRequest {
+    /// Computes the hash of the transaction.
+    pub fn hash(&self) -> B256 {
```

Comment:
It'd be good to add a test vector for this

Reply:
I think this might specifically have to do with blobs as the raw encoded transaction does not match the RLP format that we're validating. I'll skip this for now until we find something that matches our use case more accurately